pilot/pkg/model/service.go

```diff
@@ -481,8 +492,16 @@ type IstioEndpoint struct {
 	// Labels points to the workload or deployment labels.
 	Labels labels.Instance
 
-	// Address is the address of the endpoint, using envoy proto.
-	Address string
+	// Addresses is the address of the endpoint, using envoy proto:
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/endpoint/v3/endpoint_components.proto#config-endpoint-v3-endpoint-additionaladdress
+	// This change can support multiple addresses for an endpoint, then there are 2 use cases for it, such as
+	// 1. An endpoint can have both ipv4 or ipv6.
+	// 2. An endpoint can be represented a serviceentry/workload instance with multiple IP addresses.
+	// There should be some constraints below:
+	// 1. Each address of the endpoint must have the same metadata.
+	// 2. The IstioEndpoint key generated by its function Key() should be the unique identifier.
+	// Based 2, [127.0.0.1, ::1] and [::1, 127.0.0.1] should be the same instance of IstioEndpoint.
```

Comment:
We should not allow this

Reply:
Kubernetes doesn't either; the order is strict.

Reply:
Hi @howardjohn, got it, I also change the comments here.